core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -78,7 +88,7 @@ private[deploy] class Master(
   private val addressToApp = new HashMap[RpcAddress, ApplicationInfo]
   private val completedApps = new ArrayBuffer[ApplicationInfo]
   private var nextAppNumber = 0
-  private val appIdToUI = new HashMap[String, SparkUI]
+  private val appIdToUI = new ConcurrentHashMap[String, SparkUI]
```

Comment:
I was hesitant to do this because that would mean adding the `SparkUI` to a message and sending the object through the RPC layer, and I wasn't sure if it would be copied or serialized there.  If the event logs are large then that would be a lot of copying.  Is this not a problem since it's only being sent to itself?